testplan/web_ui/testing/src/AssertionPane/Assertion.js

```diff
@@ -109,79 +79,77 @@ class Assertion extends Component {
+          filter={props.filter}
+          reportUid={props.reportUid}
+          displayPath={props.displayPath}
+        />
+      );
+      break;
+    case "Summary":
+      assertionType = (
+        <SummaryBaseAssertion
+          assertion={props.assertion}
+          assertionGroupUid={props.uid}
+          filter={props.filter}
+        />
+      );
+      break;
+    default: {
+      const AssertionTypeComponent = assertionComponent(assertionType);
+      if (AssertionTypeComponent) {
         assertionType = (
-          <SummaryBaseAssertion
-            assertion={this.props.assertion}
-            assertionGroupUid={this.props.uid}
-            filter={this.props.filter}
+          <AssertionTypeComponent
+            assertion={props.assertion}
+            reportUid={props.reportUid}
           />
         );
-        break;
-      default: {
-        const AssertionTypeComponent = this.assertionComponent(assertionType);
-        if (AssertionTypeComponent) {
-          assertionType = (
-            <AssertionTypeComponent
-              assertion={this.props.assertion}
-              reportUid={this.props.reportUid}
-            />
-          );
-        } else {
-          assertionType = <NotImplementedAssertion />;
-        }
+      } else {
+        assertionType = <NotImplementedAssertion />;
       }
     }
+  };
 
-    return (
-      <Card className={css(styles.card)}>
-        <AssertionHeader
-          assertion={this.props.assertion}
-          uid={this.props.uid}
-          toggleExpand={this.props.toggleExpand}
-          index={this.props.index}
-          displayPath={this.props.displayPath}
-        />
-        <Collapse
-          isOpen={this.props.expand === EXPAND_STATUS.EXPAND}
-          className={css(styles.collapseDiv)}
-          style={{ paddingRight: isAssertionGroup ? null : "1.25rem" }}
-        >
-          <ErrorBoundary>
-            <CardBody
-              className={css(
-                isAssertionGroup
-                  ? styles.groupCardBody
-                  : styles.assertionCardBody
-              )}
-            >
-              {this.props.expand === EXPAND_STATUS.EXPAND
-                ? assertionType
-                : null}
-            </CardBody>
-          </ErrorBoundary>
-        </Collapse>
-      </Card>
-    );
-  }
+  return (
+    <Card className={css(styles.card)}>
+      <AssertionHeader
+        assertion={props.assertion}
+        uid={props.uid}
+        toggleExpand={props.toggleExpand}
+        index={props.index}
+        displayPath={props.displayPath}
+      />
+      <Collapse
+        isOpen={props.expand === EXPAND_STATUS.EXPAND}
+        className={css(styles.collapseDiv)}
+        style={{ paddingRight: isAssertionGroup ? null : "1.25rem" }}
+      >
+        <ErrorBoundary>
+          <CardBody
+            className={css(
+              isAssertionGroup
+                ? styles.groupCardBody
+                : styles.assertionCardBody
```

Comment:
While we are at refactoring a question (not keen on doing anything about it independent of the answer): are the best practices around CSS? Initially, I was used to external CSS, but here we use aphrodite (if I am not mistaken) to turn styles objects into CSS that is passed to the components.

Reply:
We can solve that under a separate task.